backend/src/campaigns.rs

```diff
@@ -157,3 +158,44 @@ pub async fn roles(
 
     Ok(Json(RolesResponse { roles }))
 }
+
+#[post("/<campaign_id>/roles", data = "<role>")]
+pub async fn create_role(
+    campaign_id: i32,
+    mut role: Form<CreateRoleInput>,
```

Comment:
No need to make role a mutable structure 

let role (line 186) should override the variable name for the compiler. - avoid mutability in rust to maintain memory safety

Reply:
The mutability is just there to use .take on one of the fields - a tiny bit faster but ultimately negligible so I will change it.